client/components/eventCard.jsx

```diff
@@ -94,7 +95,7 @@ const EventCard = (user) => {
                   <Button
                     label='Join'
                     icon={<Add />}
-                    onClick={() => subscribe(event.id, user)}
+                    onClick={() => subscribe(user.user, event.id)}
```

Comment:
subscribe is now receiving the required data for joinEvents Api request. The data seems to go no further than there...no response from the server.